apps/web/src/app/share/_components/url-shortner-form.tsx

```diff
@@ -0,0 +1,173 @@
+'use client';
+import * as React from 'react';
+import { Input } from '@repo/ui/components/input';
+import { Button } from '@repo/ui/components/button';
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@repo/ui/components/form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { useToast } from '@repo/ui/components/use-toast';
+import { createShortURLWithSlug } from '../_actions/create-url-with-slug';
+import { createShortURL } from '../_actions/create-url';
+import { Clipboard, ExternalLink } from '@repo/ui/icons';
+import { getBaseUrl } from '~/utils/getBaseUrl';
+
+const FormSchema = z.object({
+  url: z.string().url({
+    message: 'Please enter a valid URL e.g. https://typehero.dev',
+  }),
+  slug: z.string().optional(),
+});
+
+export function URLShortnerForm() {
+  const [shortURL, setShortURL] = React.useState<string | null>(null);
+  const { toast } = useToast();
+  const baseURL = `${getBaseUrl()}/share`;
+  const form = useForm<z.infer<typeof FormSchema>>({
+    mode: 'onChange',
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      url: '',
+      slug: '',
+    },
+  });
+
+  const handleShareClick = async () => {
+    if (navigator.clipboard) {
+      const url = window.location.href;
+      await navigator.clipboard.writeText(url);
+      toast({
+        variant: 'success',
+        description: 'Link To Short URL Copied!',
+        title: 'Copied',
+      });
+    }
+  };
+
+  async function onSubmit(data: z.infer<typeof FormSchema>) {
+    setShortURL(null);
+    if (data.slug && data.slug.length > 0) {
+      const shortURL = await createShortURLWithSlug(data.url, data.slug);
+      if (shortURL) {
+        toast({
+          title: 'Short URL created',
+          description: shortURL,
+          variant: 'success',
+        });
+        setShortURL(shortURL);
+      } else if (shortURL === null) {
+          toast({
```

Comment:
if it exists we should return it and display it vs showing an error cause now its a deadend